jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java

```diff
@@ -122,14 +122,33 @@ private void addRoute(KubernetesListBuilder listBuilder, ServiceBuilder serviceB
     }
 
     private static RoutePort createRoutePort(ServiceBuilder serviceBuilder) {
+
         RoutePort routePort = null;
-        final Integer serviceTargetPort = getTargetPortToExpose(serviceBuilder);
-        final Integer servicePort = serviceTargetPort != null ? serviceTargetPort : getPortToExpose(serviceBuilder);
-        if (servicePort != null) {
-            routePort = new RoutePort();
+        final ServicePort servicePortToExpose = getServicePortToExpose(serviceBuilder);
```

Comment:
Is it possible to add some unit tests verifying this precedence order?

Reply:
+1 for unit tests